### PR TITLE
refactor: drop support for etcd v2 API

### PR DESCRIPTION
**Pull request by tzssangglass**
etcd will soon drop the V2 API: https://github.com/etcd-io/etcd/issues/12913

### CODE REVIEW

**Review by membphis — APPROVED** [review]
LGTM

File: lib/resty/etcd.lua [hunk 1]
```diff
@@ -46,30 +99,28 @@ function _M.new(opts)
...
-        -- if opts special the api_prefix,no need to check version
-        if not opts.api_prefix or not utils.has_value(prefix_v3, opts.api_prefix) then
-            local ver, err = etcd_version(opts)
-            if not ver then
-                return nil, err
-            end
-            local sub_ver = ver.etcdserver:sub(1, 4)
-            opts.api_prefix = prefix_v3[sub_ver] or "/v3beta"
+    if not opts.api_prefix or not utils.has_value(prefix_v3, opts.api_prefix) then
+        local ver, err = etcd_version(opts)
```
Comment (spacewander) [inline lib/resty/etcd.lua:115]: We don't need to detect the version. The only supported prefix now is `/v3`, see https://github.com/api7/lua-resty-etcd/blob/70b648fbfb7a70f6bb257a41350fbc780a7ff58f/lib/resty/etcd.lua#L9

File: t/v3/tls.t [hunk 2]
```diff
@@ -56,7 +56,7 @@ __DATA__
             local etcd, err = require "resty.etcd" .new({
                 protocol = "v3",
```
[the remark this answers is not in the document]
Comment (tzssangglass) [reply t/v3/tls.t:57]: Maybe etcd will have a v4 protocol in the future?

**Review by spacewander — APPROVED** [review]

File: t/v3/tls.t (hunk 2 above)
[the remark this answers is not in the document]
Comment (membphis) [reply t/v3/tls.t:57]: for `protocol` field, the default value should be `v3`, all right?

**Comment by tzssangglass** [comment]
@membphis pls review again, thanks

**Pull request merged**